### PR TITLE
docs: refactored docs to match latest spec

### PR DESCRIPTION
**Pull request by semoal**
Will close some issues listed here:

#415 
#527 
#484 
#388

### CODE REVIEW

**Comment by semoal** [comment]
Not finished yet, I'll finish in 1-2 hours max: 

- [ ] Check how missing param is working on v3 and replace the current doc
- [ ] Check dynamic loading of translations of v3 and chang
- [ ] Check Webpack loader docus
- [ ] Trans misspellings

File: docs/ref/react.rst [hunk 1]
```diff
@@ -325,10 +325,16 @@ for debugging:
...
+   import { en } from 'make-plural/plurals'
+
+   i18n.loadLocaleData('en', { plurals: en })
+   i18n.load('en', messages)
+   i18n.activate('en')
 
    const App = ({ language} ) => {
         return (
-            <I18nProvider language={language} missing="🚨">
+            <I18nProvider i18n={i18n}>
```
Comment (tricoder42) [inline docs/ref/react.rst:337]: I believe `missing` prop has been removed completely and this whole paragraph can be removed as well.

Comment (tricoder42) [reply docs/ref/react.rst:337]: Ah, nevermind, I see it's still in your todo list 👍

Comment (semoal) [reply docs/ref/react.rst:337]: Yes! :) I'll ping you once finished

**Comment by semoal** [comment]
Screenshot 2020-10-20 at 16 39 34
Built a cool CRA with all the functions that Lingui uses, I'll rewrite to handle also a Hoc for dynamic loading.

@tricoder42 how is it handled on v3? Should I still use @lingui/loader?
Can we use i18n.on("change", () => {})?

File: examples/create-react-app/src/i18n.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+export const locales = {
+  en: "English",
+  cs: "Česky",
+};
+
+export const defaultLocale = "en";
+
+export const i18n = setupI18n();
+i18n.load({
```
Comment (semoal) [inline examples/create-react-app/src/i18n.ts:15]: @tricoder42 can't  we load plurals on this method?

**Comment by tricoder42** [comment]
Take a look at next.js example https://github.com/lingui/js-lingui/blob/next/examples/next-js/lingui-example/i18n.ts. I'm using a custom function, which dynamicaly loads catalog, feeds i18n object and activates locale. I believe you could use `@lingui/loader` inside the dynamic import without problems.

Also, feel free to replace create-react-app example with your version.

File: examples/create-react-app/src/i18n.ts (hunk 2 above)
Comment (tricoder42) [reply examples/create-react-app/src/i18n.ts:15]: No, this only handles messages.

Also, please don't use `setupI18n` in tutorial. That's for advanced use and it requires additional configuration in `lingui.config.js`. Simply import `i18n` from `@lingui/core`

Comment (semoal) [reply examples/create-react-app/src/i18n.ts]: Great, didn't know that I'll add that to the documentation

**Comment by semoal** [comment]
@tricoder42 Pull request ready mate :)

There's only one question left:

```jsx
  date(value: string | Date, format: Intl.DateTimeFormatOptions): string {
    return date(this.locales || this.locale, format)(value)
  }

  number(value: number, format: Intl.NumberFormatOptions): string {
    return number(this.locales || this.locale, format)(value)
  }
```

I saw this methods options are not optional, but I think it should be optional if you're conformtable with base config of i18n methods, no?

**Comment by tricoder42** [comment]
@semoal Yes, they should definitely be optional. Good point 👍 

And thanks for the PR! I'm gonna check it later tonight. Good work 🚀

**Comment by semoal** [comment]
> @semoal Yes, they should definitely be optional. Good point 👍
> 
> And thanks for the PR! I'm gonna check it later tonight. Good work 🚀

I'll create other pull request to keep the versioning tags =)

**Comment by tricoder42** [comment]
It looks perfect. Let's merge it so we can review it online. Thank you for tackling this 🚀

**Pull request merged**